src/github-client-finder/repository-finder.ts

```diff
@@ -0,0 +1,103 @@
+import * as semver from "semver";
+import * as github from "@actions/github";
+
+import { ClientDetailOverrides, InstallableClient } from "../util/types";
+import { findMatchingVersion } from "../util/version-utils";
+import {
+    getBetterHttpError, assertNotDefined, getPat,
+} from "../util/utils";
+
+export async function findClientVersionFromGithub(client: InstallableClient, desiredVersionRange: semver.Range):
+    Promise<string> {
+    const availableVersions = await findAvailableVersionFromGithub(client);
+    const clientMatchedVersion = findMatchingVersion(client, availableVersions, desiredVersionRange);
+
+    return clientMatchedVersion;
+}
+
+/**
+ * Find available versions from the github releases
+ * @returns list of the available versions for the provided client
+ */
+export async function findAvailableVersionFromGithub(client: InstallableClient):
+    Promise<string[]> {
+    const githubRepositoryPath = ClientDetailOverrides[client]?.github?.repoSlug;
+    assertNotDefined(githubRepositoryPath);
+    const slashIndex = githubRepositoryPath.indexOf("/");
+    const owner = githubRepositoryPath.substring(0, slashIndex);
+    const repo = githubRepositoryPath.substring(slashIndex + 1);
+
+    // using github token to avoid api rate limit hit
+    const octokit = github.getOctokit(getPat());
+    let releaseListresponse;
+
+    // API Documentation: https://docs.github.com/en/rest/reference/repos#list-releases
+    // Octokit Documenation: https://octokit.github.io/rest.js/v18#repos-list-releases
+    try {
+        releaseListresponse = await octokit.rest.repos.listReleases({
+            owner,
+            repo,
+        });
+    }
+    catch (err) {
+        throw getBetterHttpError(err);
+    }
+
+    const availableVersions: string[] = releaseListresponse.data.reduce(
```

Comment:
i think this maps a release to its tag_name, so it should use `map` instead of `reduce` ?